core/src/main/java/io/grpc/internal/ManagedChannelServiceConfig.java

```diff
@@ -354,9 +356,22 @@ private static RetryPolicy retryPolicy(Map<String, ?> retryPolicy, int maxAttemp
           "backoffMultiplier must be greater than 0: %s",
           backoffMultiplier);
 
+      Long perAttemptRecvTimeout =
+          ServiceConfigUtil.getPerAttemptRecvTimeoutNanosFromRetryPolicy(retryPolicy);
+      checkArgument(
+          perAttemptRecvTimeout == null || perAttemptRecvTimeout >= 0,
+          "perAttemptRecvTimeout cannot be negative: %s",
+          perAttemptRecvTimeout);
+
+      Set<Code> retryableCodes =
+          ServiceConfigUtil.getRetryableStatusCodesFromRetryPolicy(retryPolicy);
+      checkArgument(
+          perAttemptRecvTimeout != null || !retryableCodes.isEmpty(),
```

Comment:
No new tests? Seems we need tests for these cases.

Reply:
Why do we need tests for assertions. They will throw if the app is in wrong state and people will notice.

Reply:
Say if I have a bug here and the code is like
 ```
checkArgument(
      perAttemptRecvTimeout != null && !retryableCodes.isEmpty()
```
with `&&` not `||`. Then any existing code going through this code path will throw.

Reply:
> Then existing code going through this code path would throw.

The unit tests don't look like they would fail. If we can pass one or the other, it seems there should be tests for passing one or the other. I wouldn't be such a stickler but this is for service config and a bug in the logic can poison the well.

Reply:
Done.